### ISSUE TITLE
IgnorePlugin doesn't check contextRegExp for context-module-factory

### ISSUE THREAD

**Issue by anta-semenov** [webpack/webpack#4415]
**Do you want to request a *feature* or report a *bug*?**
Bug

**What is the current behavior?**

When I pass contextRegExp to IgnorePlugin constructor I'm sure that it will ignore only modules from this context, but it's not true.

**If the current behavior is a bug, please provide the steps to reproduce.**

I revealed this bug when I was trying to solve problem with skipping some locales from moment js (some of them I need to use).
I use requestRegExp: "/\.\/(?!ru|en|zh).*/" and contextRegExp: "/.*\/node_modules\/moment\/locale$/".
But the resulting bundle doesn't include content of ru.js, en.js and so on.

The problem was that context-module request './locale' passed the requestRegExp and in spite of its context (node_modules/moment) didn't pass the contextRegExp, it was ignored and all its submodules (af, ru, en...) as well.

In my case problem was solved by either adding "|locale" to requestRegExp or adding contextRegExp check to context-module-factory hook.

**What is the expected behavior?**

It should check contextRegExp if it is provided in both cases: normal-module-factory and context-module-factory.

**If this is a feature request, what is motivation or use case for changing the behavior?**

**Please mention other relevant information such as the browser version, Node.js version, webpack version and Operating System.**
webpack 2.2.1

**Comment by TheLarkInn** [comment]
Wow I think you are absolutely right. Looking into the source code for this I discovered that we are checking for the contextRegExp argument during the normal module factory hook but not the context-module-factory hook: 

```
/*
	MIT License http://www.opensource.org/licenses/mit-license.php
	Author Tobias Koppers @sokra
*/
"use strict";

class IgnorePlugin {
	constructor(resourceRegExp, contextRegExp) {
		this.resourceRegExp = resourceRegExp;
		this.contextRegExp = contextRegExp;
	}

	apply(compiler) {
		const resourceRegExp = this.resourceRegExp;
		const contextRegExp = this.contextRegExp;
		compiler.plugin("normal-module-factory", (nmf) => {
			nmf.plugin("before-resolve", (result, callback) => {
				if(!result) return callback();





                                //CHECK FOR CONTEXTMODULEFACTORY REGEXP
				if(resourceRegExp.test(result.request) &&
					(!contextRegExp || contextRegExp.test(result.context))) { 
					return callback();
				}
				return callback(null, result);
			});
		});
		compiler.plugin("context-module-factory", (cmf) => {
			cmf.plugin("before-resolve", (result, callback) => {
				if(!result) return callback();
				if(resourceRegExp.test(result.request)) {
					return callback();
				}
				return callback(null, result);
			});
		});
	}
}

module.exports = IgnorePlugin;
```


@anta-semenov are you interested in submitting a PR for this I think that it is as simple as adding that same logic line under both factory hooks!!!

**Comment by Dean-Coakley** [comment]
I opened a pull request https://github.com/webpack/webpack/pull/4418
I hope this is not considered rude. (Maybe I should have let the issue creator fix if he wants.)

If so, my apologies. First contribution!

**Comment by TheLarkInn** [comment]
Oh no that is fine!!! An example/case in configCases would be required for the PR and OP can helpful in giving you a usecase!

**Comment by timse** [comment]
fixed via #4669

**Comment by mistrykaran91** [comment]
I'm having an angular app which is quiet big, so I'm lazily loading chunks and just want to ignore few chunks. So, using IgnorePlugin but unable to match the specific folder/directory. Currently my folder structure is  src/app/modules/platform, so I want to ignore platform folder in ng-build but unable to achieve it. Any help will be appreciated. Thanks in advance.
